### PR TITLE
Add low-power support to the e-lock demo application

### PR DESCRIPTION
**Pull request by doru91**
Signed-off-by: Doru Gucea <doru-cristian.gucea@nxp.com>

#### Problem
What is being fixed?  Examples:
NXP E-lock Demo Application was lacking low power support

#### Change overview
Add low power support for the E-Lock Demo Application

#### Testing
How was this tested? (at least one bullet point required)
* Manually tested and measured the power consumption (<10 uA in low power mode),

### CODE REVIEW

**Comment by doru91** [comment]
Could someone upload the new Docker image (https://github.com/project-chip/connectedhomeip/pull/9120) to the Docker repository?  @woody-apple @andy31415

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @Damian-Nordic @msandstedt  @mspang ?

File: examples/lock-app/k32w/main/AppTask.cpp [hunk 1]
```diff
@@ -623,9 +665,21 @@ void AppTask::PostLockActionRequest(int32_t aActor, BoltLockManager::Action_t aA
...
+            if (!xQueueSendToFrontFromISR(sAppEventQueue, aEvent, &taskToWake))
+            {
+                K32W_LOG("Failed to post event to app task event queue");
+            }
+            if (taskToWake)
+            {
+                portYIELD_FROM_ISR(taskToWake);
+            }
+        }
+        else if (!xQueueSend(sAppEventQueue, aEvent, 0))
```
Comment (msandstedt) [inline examples/lock-app/k32w/main/AppTask.cpp:682]: Do we know why this previously was called with xTicksToWait=1?

File: src/platform/K32W/ble_function_mux.c [hunk 2]
```diff
@@ -177,164 +171,33 @@ uint16_t GattDb_GetIndexOfHandle(uint16_t handle)
     return gGattDbInvalidHandleIndex_d;
 }
 
-void App_NvmInit(void)
```
Comment (msandstedt) [inline src/platform/K32W/ble_function_mux.c:180]: Why are we removing the Nvm functionality?  Is it just not needed?

File: examples/lock-app/k32w/main/main.cpp [hunk 3]
```diff
@@ -137,3 +187,151 @@ extern "C" void main_task(void const * argument)
...
+}
+
+/**
+ * Glue function called directly by the OpenThread stack
+ * when system event processing work is pending.
+ */
+extern "C" void otSysEventSignalPending(void)
+{
+    /* make sure that 15.4 radio is initialized before waking up the Thread task */
+    if (dualModeStates.threadInitialized == TRUE)
```
Comment (msandstedt) [inline examples/lock-app/k32w/main/main.cpp:323]: This function is almost the same as the original version in `src/platform/K32W/ThreadStackManagerImpl.cpp`.

Why do we need this separate version with `if (dualModeStates.threadInitialized == TRUE)`?  Is that logic not valid in both builds?  If not, is there a way to organize this differently?  It's confusing that this function is in a different file based on whether we are in a low-power build.

File: src/platform/K32W/Logging.cpp [hunk 4]
```diff
@@ -120,7 +120,7 @@ void GenericLog(const char * format, va_list arg, const char * module, uint8_t c
     VerifyOrDie(writtenLen > 0);
     memcpy(formattedMsg + prefixLen + writtenLen, EOL_CHARS, EOL_CHARS_LEN);
 
-    K32WWriteBlocking((const uint8_t *) formattedMsg, strlen(formattedMsg));
+    PRINTF("%s", formattedMsg);
```
Comment (msandstedt) [inline src/platform/K32W/Logging.cpp:123]: What does this change do?

File: examples/platform/k32w/app/support/FreeRtosHooks.c [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+#if DBG_PostStepTickAssess
+    tickless_SystickCheckDriftInit();
+#endif
+
+    /* configure module for tickless mode */
+    tickless_init();
+
+#if gApp_SystemClockCalibration
+    /* start calibration - suppose 32MHz crystal is running and FRO48M running */
```
Comment (msandstedt) [inline examples/platform/k32w/app/support/FreeRtosHooks.c:113]: This comment is not very clear, and I think is also possibly inaccurate.  Are we calibrating for the ~32Khz oscillator?  That's not a crystal, and not 32MHz.

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

File: examples/platform/k32w/app/support/FreeRtosHooks.c (hunk 5 above)
Comment (doru91) [reply examples/platform/k32w/app/support/FreeRtosHooks.c]: Agree, I also updated the comment. Thanks!

**Pull request merged**